lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -0,0 +1,256 @@
+"""
+Tests for the Tool Meister modules.
+"""
+import io
+import logging
+import pathlib
+import shutil
+import subprocess
+from http import HTTPStatus
+
+import pytest
+import responses
+
+from pbench.agent.tool_meister import ToolMeister, ToolMeisterError
+
+tar_file = "test.tar.xz"
+
+
+@pytest.fixture()
+def mock_tar(monkeypatch, agent_setup):
+    def fake_run(*args, **kwargs):
+        assert kwargs["cwd"] == agent_setup["tmp"].parent
+        assert kwargs["stdin"] is None
+        assert kwargs["stderr"] == subprocess.STDOUT
+        assert kwargs["stdout"] == subprocess.PIPE
+        c = subprocess.CompletedProcess(args, returncode=0, stdout=b"", stderr=None)
+        assert all(
+            x in c.args[0]
+            for x in [
+                "tar_path",
+                "--create",
+                "--xz",
+                "--force-local",
+                f"--file={tar_file}",
+            ]
+        )
+        return c
+
+    monkeypatch.setattr(subprocess, "run", fake_run)
+
+
+@pytest.fixture(scope="function")
+def logger_fixture(request):
+    logger = logging.getLogger(f"__logger__{request.node.name}")
```

Comment:
How does this make the logger unique for the test?

Perhaps just use a UUID?

Reply:
So that is a bit obscure.  =)

Let's add a comment to make sure that assumption is understood clearly.